app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -281,20 +330,98 @@ fun GamblingScreen(navigationActions: NavigationActions, userViewModel: UserView
+                              style = MaterialTheme.typography.bodyMedium,
+                              modifier = Modifier.testTag("current_level_text"))
+                          Text(
+                              text = stringResource(R.string.level_display, (displayedLevel + 1)),
+                              style = MaterialTheme.typography.bodyMedium,
+                              modifier = Modifier.testTag("next_level_text"))
+                        }
+
+                    LinearProgressIndicator(
+                        progress = { animatedProgress.value },
+                        modifier =
+                            Modifier.fillMaxWidth(0.8f).height(8.dp).testTag("level_progress_bar"),
+                        color = MaterialTheme.colorScheme.primary,
+                    )
+
+                    Spacer(modifier = Modifier.height(8.dp))
+
+                    AnimatedVisibility(
+                        visible = displayedLevel < flooredLevel,
+                        enter = fadeIn() + expandVertically(),
+                        exit = fadeOut() + shrinkVertically()) {
+                          Text(
+                              text = stringResource(R.string.level_up),
+                              color = MaterialTheme.colorScheme.primary,
+                              style = MaterialTheme.typography.headlineSmall,
+                              modifier = Modifier.padding(vertical = 8.dp).testTag("level_up_text"))
+                        }
+
+                    Text(
+                        text = stringResource(R.string.gambling_screen_coins_display, coins ?: 0),
+                        modifier = Modifier.testTag("coin_display"),
+                        style = MaterialTheme.typography.headlineMedium)
+
+                    AnimatedVisibility(
+                        visible = showXpIncrement && (xpIncrement > 0),
+                        enter = fadeIn(animationSpec = tween(durationMillis = 500)),
+                        exit = fadeOut(animationSpec = tween(durationMillis = 500))) {
+                          Text(
+                              text = stringResource(R.string.xp_increment, rarityText, xpIncrement),
+                              style = MaterialTheme.typography.bodyMedium.copy(color = Color.Green),
+                              modifier =
+                                  Modifier.align(Alignment.CenterHorizontally)
+                                      .padding(top = 16.dp)
+                                      .testTag("xp_increment_text"))
+                        }
+                  }
 
               WheelView(
                       modifier = Modifier.size(300.dp).testTag("wheel_view"),
-                      onSegmentLanded = { /* Handle landing */})
+                      onSegmentLanded = { segmentName ->
+                        val reputationIncrement =
+                            when (segmentName) {
```

Comment:
Here to avoid "code duplication" you could create a val for the normalization factor `sqrt(currentLevel + 1.0)`

Reply:
indeed